Mathlib/MeasureTheory/Function/FactorsThrough.lean

```diff
@@ -37,14 +40,90 @@ theorem _root_.Measurable.factorsThrough [MeasurableSpace Z] [MeasurableSingleto
 /-- If a function `g` is strongly measurable with respect to the pullback along some function `f`,
 then to prove `g x = g y` it is enough to prove `f x = f y`.
 
-TODO: under certain assumptions, the factorization map `h` is measurable. This is the content of the
-[Doob-Dynkin lemma](https://en.wikipedia.org/wiki/Doob–Dynkin_lemma). -/
+Under certain assumptions, the factorization map `h` is measurable
+(see `exists_eq_measurable_comp`). -/
 theorem StronglyMeasurable.factorsThrough [TopologicalSpace Z]
     [PseudoMetrizableSpace Z] [T1Space Z] (hg : StronglyMeasurable[mY.comap f] g) :
     g.FactorsThrough f := by
   borelize Z
   exact hg.measurable.factorsThrough
 
+variable {ι : Type*} [MetricSpace Z] [CompleteSpace Z] [Countable ι] {l : Filter ι}
+  [l.IsCountablyGenerated] {f : ι → X → Z}
+
+theorem StronglyMeasurable.measurableSet_exists_tendsto [MeasurableSpace X]
```

Comment:
I'm not sure this lemma and the next one belong to this file: they are general results on strongly measurable functions.